src/client.ts

```diff
@@ -2040,6 +2085,16 @@ export class StreamChat<
    */
   _isUsingServerAuth = () => !!this.secret;
 
+  _isRunningNode = () => {
+    let isNode;
+    try {
+      isNode = Object.prototype.toString.call(global.process) === '[object process]';
```

Comment:
Looks too complicated IMO. Cant we just check if `global.process` is defined instead of catching an exception?

Reply:
This is supposed to be a very reliable way to do it. Source: https://github.com/iliakan/detect-node.